src/components/Table/Table.tsx

```diff
@@ -170,20 +217,50 @@ export const Table: ChakraComponent<
         }
       }
 
+      const wapperStyles = isScrollable
+        ? {
+            overflow: isScrollable ? "auto" : "hidden",
+            maxWidth: "100%",
+            /** Show shadow to scroll */
+            background:
+              "linear-gradient(to right, white 30%, rgba(255,255,255,0)), linear-gradient(to right, rgba(255,255,255,0), white 70%) 0 100%, radial-gradient(farthest-side at 0% 50%, rgba(0,0,0,.2), rgba(0,0,0,0)), radial-gradient(farthest-side at 100% 50%, rgba(0,0,0,.2), rgba(0,0,0,0)) 0 100%",
+            backgroundRepeat: "no-repeat",
+            backgroundColor: "white",
+            backgroundSize: "40px 100%, 40px 100%, 14px 100%, 14px 100%",
+            backgroundPosition: "0 0, 100%, 0 0, 100%",
+            backgroundAttachment: "local, local, scroll, scroll",
+          }
+        : undefined;
+
+      const wrapperProps = isScrollable
+        ? {
+            overflow: "auto",
+            role: "region",
+            tabIndex: 0,
+            whiteSpace: "wrap",
+          }
+        : {
+            overflow: "hidden",
+            whiteSpace: "wrap",
+          };
+
```

Comment:
Can these rules be moved to the theme object and then assigned as `<TableContainer sx={styles}>`?

Reply:
These styles and attributes are for a conditional parent of the actual table element.  The parent is not always applied to the component.  Is there a way to set styles for a parent element from within the `Table` theme object?

Reply:
What is the conditional parent? Is it `TableContainer` and why is it not always applied to the component?

Reply:
You can/should think of the base theme object styles, the outermost object, as the parent. If it always renders then it would get `sx={styles}` as the prop. `ChakraTable` would need something like `sx={styles.innerTable}` (random name).

Reply:
I was able to rework the styles as a multi-part configuration.  :)